apps/backend/src/bookings/bookings.service.ts

```diff
@@ -0,0 +1,332 @@
+        booking.bookingStatus !== 'TRAVELING'
+      ) {
+        throw new UnprocessableEntity(
+          'this booking has been accepted or cancelled',
+        );
+      }
+
+      const cancelledBooking = await this.bookingsRepo.updateBookingStatus(
+        bookingId,
+        'GUIDE_CANCELLED',
+      );
+
+      if (booking.bookingStatus === 'TRAVELING') {
+        await this.paymentsService.refund(bookingId);
+        return {
+          id: cancelledBooking.id,
+          refunded: true,
+          bookingStatus: cancelledBooking.bookingStatus.toString(),
+        };
+      }
+
+      return {
+        id: cancelledBooking.id,
+        refunded: false,
+        bookingStatus: cancelledBooking.bookingStatus.toString(),
+      };
+    } catch (e) {
+      if (e instanceof GuideNotFound) {
+        throw new AccessNotGranted('permission denied');
+      }
+      throw e;
+    }
+  }
+
+  async payBookingFee(
+    bookingId: number,
+    account: AccountMetadata,
+    token: string,
+  ): Promise<PayBookingFeeResponse> {
+    try {
+      const booking = await this.bookingsRepo.getBookingById(bookingId);
+      if (booking.userId !== account.userId) {
+        throw new AccessNotGranted('permissing denied');
+      }
+      if (booking.bookingStatus !== 'WAITING_FOR_PAYMENT') {
+        throw new UnprocessableEntity(
+          'this booking has not beed waiting for payment',
+        );
+      }
+
+      const now = new Date();
+      const paymentDeadline = moment(booking.updatedAt).add(1, 'days').toDate();
+      if (now > paymentDeadline) {
+        const cancelledBooking = await this.bookingsRepo.updateBookingStatus(
+          bookingId,
+          'USER_CANCELLED',
+        );
+        return {
+          message: 'booking payment exceeds the deadline',
+          bookingId: cancelledBooking.id,
+          bookingStatus: cancelledBooking.bookingStatus.toString(),
+        };
+      }
+
+      await this.paymentsService.charge({
+        userId: account.userId,
+        bookingId: bookingId,
+        token: token,
+      });
+
+      const paidBooking = await this.bookingsRepo.updateBookingStatus(
+        bookingId,
+        'TRAVELING',
+      );
+      return {
+        message: 'success',
+        bookingId: paidBooking.id,
+        bookingStatus: paidBooking.bookingStatus.toString(),
+      };
+    } catch (e) {
+      if (e instanceof GuideNotFound) {
+        throw new AccessNotGranted('permission denied');
+      }
+      throw e;
+    }
+  }
+
+  // TRAVELING = PAID
+  async cancelBookingByTraveller(
+    bookingId: number,
+    account: AccountMetadata,
+  ): Promise<CancelBookingByTravellerResponse> {
+    try {
+      const booking = await this.bookingsRepo.getBookingById(bookingId);
+      console.log({ booking, account });
+      if (booking.userId !== account.userId) {
+        throw new AccessNotGranted('permissing denied');
+      }
+      if (
+        booking.bookingStatus !== 'TRAVELING' &&
```

Comment:
Does this line means the traveller can cancel the trip despite it already been started
sounds weird, but i dunno